labs/b9.md

```diff
@@ -1,159 +1,539 @@
+or code style mistakes. For projects on public repostitories that invite
+contributions from strangers, you can contribute a feature by forking the
+repository (making a copy of it that you own), implementing your feature, and
+then making a pull request to have your contribution merged into the official
+repository.
+
+First, read up on [how to create a pull
+request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request)
+from Github's documentation. We've created a dummy repository at [GITHUB URL FOR
+CURRENT SEMESTER HERE] for you to practice making a simple PR from a fork. You
+will be making a fork of the repo, writing your name in the Markdown file in the
+repo, and then making a PR to merge your change into the original repo.
+
+1. Read about [how to fork a
+   repo](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo)
+   from Github's documentation. Make a fork of our dummy repository at [GITHUB
+   URL FOR CURRENT SEMESTER HERE].
+
+2. Clone your forked repository and create a new branch based on `master` called
+   `my-name`. Open `name.md` and fill in your name.
+> Hello World! My name is Tux the Penguin.
+
+3. Stage and commit your change with the commit message `Add my name`, and then
+   `git push`.
+
+4. Read about [how to make a PR from a
+   fork](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)
+   from Github's documentation. Make a PR from your branch `my-name` on your
+   forked repo `<Your Github username>/[NAME OF DUMMY REPO]` to the branch
+   `master` on the original repo `0xcf/[NAME OF DUMMY REPO]`.
+
+You've now made a pull request! Github has a lot of features for pull requests,
+feel free to check some of them out. This lab you're reading right now was
+merged through a PR, which you can view at
+https://github.com/0xcf/decal-web/pull/216.
+
+- The main page for the pull request has a timeline of commits and any comments
+  from reviewers. PRs can be continually updated by making and pushing new
+  commits to your compare branch. Github will track any new changes pushed to
+  the compare branch and update your PR.
+
+- The sidebar on the main page has several notable features. You can request
+  other people to review your changes, assign others to work on the PR, and link
+  an issue that your PR will resolve if it gets merged. Github repositories have
+  an "Issues" tab where one can submit a report about a bug or request a feature
+  (e.g. https://github.com/0xcf/decal-web/issues). The Github issue associated
+  with the PR for this lab is at https://github.com/0xcf/decal-web/issues/184.
+
+- Go to the "Files Changed" tab to see a diff of the PR's changes. Reviewers of
+  your PR can leave inline comments on a specific line of code by hovering over
+  the line and clicking the blue "+" button that pops up.
+
```

Comment:
Suggestion: add a gradescope question here asking them to paste in their PR link.